main.cpp

```diff
@@ -0,0 +1,379 @@
+        return true;
+    }
+    
+    timesAndRepetitions[indexSelectionMode + 1] -= 1;
+    return true;
+}
+
+void decreaseRepetitions() {
+    if (timesAndRepetitions[indexSelectionMode] == 0) {
+        return;
+    }
+
+    timesAndRepetitions[indexSelectionMode] -= 1;
+    return;
+}
+
+void display() {
+    if (indexSelectionMode == 4) {
+        displayRepetitions();
+        return;
+    } 
+    displayTime();
+};
+
+void displayModeSeleted() {
+    switch (indexSelectionMode) {
+        case 0:
+            uartUsb.write(" exercise time selected\r\n",25);
+            break;
+        case 2:
+            uartUsb.write(" rest time selected\r\n",21);
+            break;
+        case 4:
+            uartUsb.write(" repetitions selected\r\n",23);
+            break;
+        default:
+            indexSelectionMode = 0;
+            break;
+    }
+
+}
+
+void displayRepetitions() {
+    char str[2];
+    sprintf(str,"%d",timesAndRepetitions[indexSelectionMode]);
+    uartUsb.write(&str,2);
+    displayModeSeleted();
+    uartUsb.write("\r\n",2);
+    uartUsb.write(" \r\n",3);
+};
+
+void displayTime() {
+    char str[2];
+    sprintf(str,"%d",timesAndRepetitions[indexSelectionMode]);
+    uartUsb.write(&str,2);
+    uartUsb.write(" : ",3);
+    sprintf(str,"%d",timesAndRepetitions[indexSelectionMode + 1]);
+    uartUsb.write(&str,2);
+    displayModeSeleted();
+    uartUsb.write("\r\n",2);
+    uartUsb.write(" \r\n",3);
+};
+
+void increase() {
+    uartUsb.write("\r\nincrement\r\n",13);
+
+    if (indexSelectionMode == 4) {
+        increaseRepetitions();
+        return;
+    } 
+    increaseTime();
+    return;
+};
+
+void increaseRepetitions() {
+    if (timesAndRepetitions[indexSelectionMode] == 99) {
+        timesAndRepetitions[indexSelectionMode] = 0;
+
+        return;
+    }
+
+    timesAndRepetitions[indexSelectionMode] += 1;
+    return;
+}
+
+bool increaseTime() {
+    if (timesAndRepetitions[indexSelectionMode + 1] == 59) {
+        if (timesAndRepetitions[indexSelectionMode] == 99) {
+        return false;
+    }
+        timesAndRepetitions[indexSelectionMode]    += 1;
+        timesAndRepetitions[indexSelectionMode + 1] = 0;
+        return true;
+    }
+    
+    timesAndRepetitions[indexSelectionMode + 1] += 1;
+    return true;
+}
+
+void manageLEDs() {
+    switch (indexSelectionMode) {
+        case 0:
+            exerciseLED   = true;
+            restLED       = false;
+            repetitionLED = false;
+            break;
+        case 2:
+            exerciseLED   = false;
+            restLED       = true;
+            repetitionLED = false;
+            break;
+        case 4:
+            exerciseLED   = false;
+            restLED       = false;
+            repetitionLED = true;
+            break;
+        default:
+            exerciseLED   = false;
+            restLED       = false;
+            repetitionLED = true;
+            break;
+    }
+}
+
+void onRXInterrupt(){

```

Comment:
Dónde estás usando esta función. Me perdí.

Reply:
El llamado a la función lo borré y me olvidé de sacar esa parte también. En principio tenía esa función y _readButtons()_ en el while así: 
```
while (true) {
        readButtons();
        onRXInterrupt();
        ThisThread::sleep_for(DEAD_TIME_FOR_DEBOUNCING);
    }
```
pero cuando incrementaba la cuenta de 9 a 10 segundos, por puerto serie salía un error tipo *HardFault exception
Cortex-M HardFault exception has occurred*.
Hice debugging pero no logré descrifrar por qué se daba el error.  Luego noté que, al quitar _onRXInterrupt()_ para que la función sólo lea los botones, no tenía error. Y como la acción de ingresar comandos no estuvo contemplada en la idea de proyecto, decidí al menos para ésta entrega dejarlo así.

Reply:
Ok, perfecto.